packages/runtime/container-runtime/src/containerRuntime.ts

```diff
@@ -2170,8 +2166,27 @@ export class ContainerRuntime extends TypedEventEmitter<IContainerRuntimeEvents>
             }
         }
 
-        this.blobManager.updateUnusedRoutes(blobManagerUnusedRoutes, tombstone);
-        this.dataStores.updateUnusedRoutes(dataStoreUnusedRoutes, tombstone);
+        this.blobManager.updateUnusedRoutes(blobManagerUnusedRoutes);
+        this.dataStores.updateUnusedRoutes(dataStoreUnusedRoutes);
+    }
+
+    /**
+     * This is called to update objects that are tombstones.
+     * @param tombstonedRoutes - Data store and attachment blob routes that are tombstones in this Container.
+     */
+    public updateTombstonedRoutes(tombstonedRoutes: string[]) {
+        const blobManagerTombstonedRoutes: string[] = [];
+        const dataStoreTombstonedRoutes: string[] = [];
+        for (const route of tombstonedRoutes) {
+            if (this.isBlobPath(route)) {
+                blobManagerTombstonedRoutes.push(route);
+            } else {
+                dataStoreTombstonedRoutes.push(route);
+            }
+        }
```

Comment:
Thought: Do we do this calculation enough times to separate it into a helper function?

Reply:
yeah maybe, let me take a look.

Reply:
Moved to helper function.